src/transformers/integrations/integration_utils.py

```diff
@@ -762,9 +762,8 @@ def setup(self, args, state, model, **kwargs):
             if trial_name is not None:
                 init_args["name"] = trial_name
                 init_args["group"] = args.run_name
-            else:
-                if not (args.run_name is None or args.run_name == args.output_dir):
-                    init_args["name"] = args.run_name
+            elif args.run_name is not None:
+                init_args["name"] = args.run_name
```

Comment:
Hi @qubvel : One suggestion. Since this is changing the expectation for some W&B users can we add the following warning?

```self._wandb.termwarn("Please set `TrainingArguments.run_name` to specify the W&B run name. If unset, `TrainingArguments.output_dir` will be used by default., repeat=False)```

Reply:
Ok, added with a bit different wording:

```python
self._wandb.termwarn(
   "The `run_name` is currently set to the same value as `TrainingArguments.output_dir`. If this was "
   "not intended, please specify a different run name by setting the `TrainingArguments.run_name` parameter.",
   repeat=False,
)
```